full/full_coupler_mod.F90

```diff
@@ -1112,9 +1153,137 @@ end subroutine coupler_init
+    type(ocean_ice_boundary_type),  target, intent(in) :: Ocean_ice_boundary  !< Ocean_ice_boundary
+
+    this%Atm => Atm
+    this%Land => Land
+    this%Ice => Ice
+    this%Ocean => Ocean
+    this%Land_ice_atmos_boundary => Land_ice_atmos_boundary
+    this%Atmos_land_boundary => Atmos_land_boundary
+    this%Atmos_ice_boundary => Atmos_ice_boundary
+    this%Land_ice_boundary => Land_ice_boundary
+    this%Ice_ocean_boundary => Ice_ocean_boundary
+    this%Ocean_ice_boundary => Ocean_ice_boundary
+
+  end subroutine initialize_coupler_components_obj
+
+  !> Function get_component returns the requested component in the coupler_components_type object
+  !! Users are required to provide the component to be retrieved as an input argument.  For example,
+  !! coupler_components_obj%get_component(Atm) will return Atm = coupler_components_obj%Atm
+  subroutine get_component(this, retrieve_component )
+
+    implicit none
+    class(coupler_components_type), intent(in) :: this !< the coupler_components_type object
+    class(*), intent(out) :: retrieve_component  !< requested component to be retrieve.
+                             !! retrieve_component can be of type atmos_data_type, land_data_type, ice_data_type,
+                             !! ocean_public_type, land_ice_atmos_boundary_type, atmos_land_boundary_type,
+                             !! atmos_ice_boundary_type, land_ice_boundary_type, ice_ocean_boundary_type,
+                             !! ocean_ice_boundary_type
+
+    select type(retrieve_component)
+    type is(atmos_data_type) ; retrieve_component = this%Atm
+    type is(land_data_type)  ; retrieve_component = this%Land
+    type is(ice_data_type)   ; retrieve_component = this%Ice
+    type is(ocean_public_type) ; retrieve_component = this%Ocean
+    type is(land_ice_atmos_boundary_type) ; retrieve_component = this%Land_ice_atmos_boundary
+    type is(atmos_land_boundary_type) ; retrieve_component = this%Atmos_land_boundary
+    type is(atmos_ice_boundary_type)  ; retrieve_component = this%Atmos_ice_boundary
+    type is(land_ice_boundary_type)   ; retrieve_component = this%Land_ice_boundary
+    type is(ice_ocean_boundary_type)  ; retrieve_component = this%Ice_ocean_boundary
+    type is(ocean_ice_boundary_type)  ; retrieve_component = this%Ocean_ice_boundary
+    class default
+      call fms_mpp_error(FATAL, "failure retrieving component in coupler_components_type object, &
+                         cannot recognize the type of requested component")
+    end select
+
+  end subroutine get_component
+
+  !> Function set_component sets the requested component in the coupler_components_type object
+  !! Users are required to provide the component to be set as an input argument.  For example,
+  !! coupler_components_obj%get_component(Atm) will set coupler_components_obj%Atm = Atm
+  subroutine set_component(this, set_this_component )
```

Comment:
I would be shocked if this worked.  I don't think these types all have overloaded `=`

Reply:
It has been removed.